### PR TITLE
hunspell: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

**Comment by CJCombrink** [comment]
Hi @valgur 
I see I created a pr for the same lib [here](https://github.com/conan-io/conan-center-index/pull/18829) :P

Which is good, It seems we came to the same solution (almost...)
Do you have any idea why there is an issue with your pr?
Also let me know if I can help in any way to get this through.

File: recipes/hunspell/all/test_package/conanfile.py [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not tools.cross_building(self):
-            bin_path = os.path.join("bin", "test_package")
-            self.run(bin_path, run_environment=True)
+        if can_run(self):
+            bin_path = os.path.join(self.cpp.build.bindir, "test_package")
```
Comment (CJCombrink) [inline recipes/hunspell/all/test_package/conanfile.py:25]: Maybe?
```suggestion
            bin_path = os.path.join(self.cpp.build.bindir[0], "test_package")
```

See my pr: [here](https://github.com/conan-io/conan-center-index/pull/18829/files#diff-d37f3885bf913ef957677a9101f6732d0b5bff277f7c3f432abee2edd7f35f49R25)

Comment (prince-chrismc) [reply recipes/hunspell/all/test_package/conanfile.py:25]: These are the same so either works... (you missed an s in your suggestion)

**Comment by valgur** [comment]
Closing in favor of #18829.

**Pull request closed without merging**